src/routes/PostDetail/Comments/Comments.js

```diff
@@ -29,6 +28,26 @@ export default class Comments extends Component {
     comments: []
   }
 
+  componentDidMount () {
+    this.ensureSelectedCommentPresent()
+  }
+
+  componentDidUpdate (prevProps) {
+    this.ensureSelectedCommentPresent()
+  }
+
+  ensureSelectedCommentPresent () {
+    const { comments, commentsPending } = this.props
+    const selectedCommentId = findCommentId(window.location.pathname)
+    const commentIds = []
+    comments.forEach(comment => {
+      commentIds.push(comment.id)
+      comment.childComments.forEach(comment => commentIds.push(comment.id))
+    })
+    if (comments.length < 1) return
+    if (!commentsPending && !commentIds.includes(selectedCommentId.toString())) this.props.fetchComments().then(() => this.forceUpdate())
```

Comment:
so weird that it wouldn't re-render. maybe something is not being passed down from the connector that should be? the comments should change and then a render should happen

Reply:
Indeed.